tests/components/landisgyr_heat_meter/test_init.py

```diff
@@ -20,3 +25,40 @@ async def test_unload_entry(hass):
     assert "landisgyr_heat_meter" in hass.config.components
 
     assert await hass.config_entries.async_remove(entry.entry_id)
+
+
+async def test_migrate_entry(hass):
+    """Test successful migration of entry data from version 1 to 2."""
+
+    # Create entry (which itself is not changed in version 2)
+    entry_data = {
+        "device": "/dev/USB0",
+        "model": "LUGCUH50",
+        "device_number": "12345",
+    }
+    entry = MockConfigEntry(
+        domain="landisgyr_heat_meter",
+        title="LUGCUH50",
+        entry_id="987654321",
+        data=entry_data,
+    )
+
+    assert entry.data == entry_data
+    assert entry.version == 1
+
+    # Create entity entry to migrate to new unique ID
+    registry = er.async_get(hass)
+    registry.async_get_or_create(
+        SENSOR_DOMAIN,
+        LANDISGYR_HEAT_METER_DOMAIN,
+        "landisgyr_heat_meter_987654321_measuring_range_m3ph",
+        suggested_object_id="heat_meter_measuring_range",
+        config_entry=entry,
+    )
+
+    await entry.async_migrate(hass)
```

Comment:
Please set up the config entry with `hass.config_entries.async_setup` instead. That will also migrate the config entry.

Reply:
No. We set the version of the config entry when creating the `MockConfigEntry`. Setting up the config entry will migrate it to the new version of the config flow.

Reply:
Mm, thanks, I'll have to research that a little I guess. The migration touches the entity unique id's, which are not part of the MockConfigEntry (at least not the way I had it).

Reply:
I'd copied the way some integration ('axis' and 'deconz') have done it, but I've found others like 'sabnzbd' that use async_setup which is looking good. Please let me know if this is what you meant. And thanks for your help and patience!